src/components/Table/Table.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react"
+import { ColumnConfig, ColumnConfigs, WithID } from "./types"
+
+interface Props<Datum extends WithID> {
+  data: Datum[]
+  columnConfigs: ColumnConfigs<Datum>
+}
+
+export function Table<Datum extends WithID>(props: Props<Datum>) {
+  const { data: dataUnsorted, columnConfigs } = props
+  const [sortConfig, setSortConfig] = useState<ColumnConfig<Datum>>()
+  const [sortAsc, setSortAsc] = useState<boolean>(false)
+
+  const data = [...dataUnsorted].sort((a, b) => {
+    const valueA = sortConfig?.getValue?.(a)
+    const valueB = sortConfig?.getValue?.(b)
+    if (isSortable(valueA) && isSortable(valueB)) {
+      if (valueA > valueB) return sortAsc ? 1 : -1
+      if (valueA < valueB) return sortAsc ? -1 : 1
+    }
+    return 0
+  })
+
+  const handleSort = (columnConfig: ColumnConfig<Datum>) => () => {
+    if (columnConfig.sortable && columnConfig.getValue) {
+      if (columnConfig === sortConfig || !sortConfig) {
+        setSortAsc((value) => !value)
+      }
+      setSortConfig(columnConfig)
+    }
+  }
```

Comment:
Any thoughts on `if` nesting? It's pretty easy to follow in this case but I find that nesting becomes hard to track when functions are on the larger side.

However, there's usually a way to invert the conditions and remove nesting:
```tsx
if (!columnConfig.sortable || !columnConfig.getValue) return;

if (columnConfig === sortConfig || !sortConfig) {
    setSortAsc((value) => !value)
}

setSortConfig(columnConfig)
```

Reply:
That's smart 👌

Reply:
@luiz-chagas I love this. I see this in your code often and really appreciate how clean this is. For some reason I always avoid empty returns, but is such a clean way of escaping the function.